### PR TITLE
add db initialization with sample data

### LINKED ISSUE #5: Intitial cards model and data population of mongoDB.
**EDIT:** Add some basic Card.java `@Document`class, that is extensible enough to serve as the base class for all our flash cards documents - i.e. the instances that make up the collection 
-> consider abstract or sealed class - but not exaggerated complexity. Derived classes could be SimpleQAndACard, MultipleChoiceCard, and may well be records)
*--- end of edit to data model --- added, since it is probably needed for the population task below ---*

Add some initial data to populate the mongoDB:

- a sample user (to the user collection)
- some initial card data - as an example collection `db.example`.

The data are well-suited for us now in early development phase, as they can be used for implementing display / modification views. But they can also stay in permanently and serve to new users to flashcards as public card sample collection to get familiar.

Incrementally, the example collection should consist of different card types as simple Q&A, multiple choice, etc.. to illustrate the options a user has.

Several alternatives - increasingly more effort - on how to implement the task (all alternative should only insert, if the example collection does not yet exist)

1. **Insert manually per script**. 
Use init script to insert data  - like 5 to 10 Json records for a start (comparable to `spring.sql.init.data-locations=classpath:initData.sql`)
2. **Insert manually via CommandLineRunner**. 
same as 1 only Commandlinerunner
3. **use quiz data from the net**.  
find and download some suitable (maybe a little larger) quiz data set and transform it into our cards example collection by use of a CommandLineRunner.
4. **scrape website and generate cards**
e.g. use the problems of some Hyperskill topic group (e.g. Spring)

### LINKED ISSUE #14: Add sample flashcards to the db
Extracted from Issue #5
Depends on Issue #13 

Add some initial data to populate the mongoDB:

- a sample user (to the user collection)
- some initial card data - as an example collection `db.example`.

The data are well-suited for us now in early development phase, as they can be used for implementing display / modification views. But they can also stay in permanently and serve to new users to flashcards as public card sample collection to get familiar.

Incrementally, the example collection should consist of different card types as simple Q&A, multiple choice, etc.. to illustrate the options a user has.

Several alternatives - increasingly more effort - on how to implement the task (all alternative should only insert, if the example collection does not yet exist)

- Insert manually per script. Use init script to insert data - like 5 to 10 Json records for a start (comparable to `spring.sql.init.data-locations=classpath:initData.sql`)
- Insert manually via CommandLineRunner. Same as 1 only Commandlinerunner.

### PR DESCRIPTION
**Pull request by alfabravo2013**

### CODE REVIEW

File: server/src/main/java/org/hyperskill/community/flashcards/card/ExampleDataInitializer.java [hunk 1]
```diff
@@ -21,63 +30,84 @@
...
+    private static final String flashcardsJsonPath = "/json/flashcards.json";
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final UserMapper userMapper;
     private final MongoTemplate mongoTemplate;
 
-    public ExampleDataInitializer(@Qualifier("example") MongoTemplate mongoTemplate) {
+    public ExampleDataInitializer(UserMapper userMapper, MongoTemplate mongoTemplate) {
+        this.userMapper = userMapper;
         this.mongoTemplate = mongoTemplate;
     }
```
Comment (wisskirchenj) [inline server/src/main/java/org/hyperskill/community/flashcards/card/ExampleDataInitializer.java]: not important. - but just in general:
you can nicely save this constructor boiler plate with `@RequiredArgsContructor`

File: server/src/main/java/org/hyperskill/community/flashcards/config/MongoConfiguration.java [hunk 2]
```diff
@@ -15,10 +14,4 @@ public class MongoConfiguration {
     public MongoTemplate mongoTemplate() {
         return new MongoTemplate(MongoClients.create(), "cards");
     }
-
-    @Bean
-    @Qualifier("example")
-    public MongoTemplate exampleMongoTemplate() {
-        return new MongoTemplate(MongoClients.create(), "example");
-    }
 }
```
Comment (wisskirchenj) [inline server/src/main/java/org/hyperskill/community/flashcards/config/MongoConfiguration.java:17]: so now you work on cards database?
I dno't mind - just wondering...

**Review by codeofcarbon — APPROVED** [review]
since I am not a mongoDb user (yet) you won't have any big upgrade suggestions from my side. 
I am able to run the code smoothly, and connect with the database without any issues, so LGTM

File: server/src/main/java/org/hyperskill/community/flashcards/card/ExampleDataInitializer.java (hunk 1 above)
Comment (alfabravo2013) [reply server/src/main/java/org/hyperskill/community/flashcards/card/ExampleDataInitializer.java]: Thanks for suggestion, done

File: server/src/main/java/org/hyperskill/community/flashcards/config/MongoConfiguration.java (hunk 2 above)
Comment (alfabravo2013) [reply server/src/main/java/org/hyperskill/community/flashcards/config/MongoConfiguration.java:17]: Yes, it populates the development `cards` database with 2 users and 12 flashcards of different types, all put in the `example` collection which belongs to the `test1@test.com` user. This is to facilitate development of endpoints and other components based on the actual database.

**Review by wisskirchenj — APPROVED** [review]

**Pull request merged**